src/docgen/transpile/csharp.ts

```diff
@@ -0,0 +1,226 @@
+      return { name: parentFqn, submodule: submoduleCSharpPackage };
+    }
+
+    return { name: csharpPackage };
+  }
+
+  public type(type: reflect.Type): transpile.TranspiledType {
+    const submodule = this.findSubmodule(type);
+    const moduleLike = this.moduleLike(submodule ? submodule : type.assembly);
+
+    const fqn = [];
+
+    let namespace = type.namespace;
+    if (namespace) {
+      fqn.push(moduleLike.name);
+      fqn.push(namespace);
+    } else {
+      fqn.push(moduleLike.name);
+    }
+    fqn.push(type.name);
+
+    return {
+      fqn: fqn.join('.'),
+      name: type.name,
+      namespace: namespace,
+      module: moduleLike.name,
+      submodule: moduleLike.submodule,
+      source: type,
+      language: this.language,
+    };
+  }
+
+  public callable(callable: reflect.Callable): transpile.TranspiledCallable {
+    const type = this.type(callable.parentType);
+    const isInitializer = callable.kind === reflect.MemberKind.Initializer;
+    const name = isInitializer
+      ? type.name
+      : Case.pascal(callable.name);
+
+    const params = callable.parameters.map(p => this.formatParameter(this.parameter(p))).join(', ');
+    const signatures = [`public ${name}(${params})`];
+    const invocations = [isInitializer
+      ? `new ${name}(${params});`
+      : `${type.name}.${name}(${params});`];
+
+    return {
+      name,
+      parentType: type,
+      import: this.formatImport(type),
+      parameters: callable.parameters,
+      signatures,
+      invocations,
+    };
+  }
+
+  public class(klass: reflect.ClassType): transpile.TranspiledClass {
+    return {
+      name: klass.name,
+      type: this.type(klass),
+    };
+  }
+
+  public struct(struct: reflect.InterfaceType): transpile.TranspiledStruct {
+    const type = this.type(struct);
+    const indent = ' '.repeat(4);
+    const inputs = struct.allProperties.map((p) =>
+      this.formatBuilderMethod(this.property(p), indent),
+    ).flat();
+    return {
+      type: type,
+      name: struct.name,
+      import: this.formatImport(type),
+      initialization: this.formatStructBuilder(type, inputs),
+    };
+  }
+
+  public interface(
+    iface: reflect.InterfaceType,
+  ): transpile.TranspiledInterface {
+    return {
+      name: iface.name,
+      type: this.type(iface),
+    };
+  }
+
+  public parameter(
+    parameter: reflect.Parameter,
+  ): transpile.TranspiledParameter {
+    const typeRef = this.typeReference(parameter.type);
+    return {
+      name: parameter.name,
+      parentType: this.type(parameter.parentType),
+      typeReference: typeRef,
+      optional: parameter.optional,
+      declaration: this.formatProperty(parameter.name, typeRef),
+    };
+  }
+
+  public property(property: reflect.Property): transpile.TranspiledProperty {
+    const typeRef = this.typeReference(property.type);
+    return {
+      name: property.name,
```

Comment:
Shouldn't this be `PascalCased`? Basing it of on https://github.com/aws-samples/aws-cdk-examples/blob/master/csharp/application-load-balancer/src/ApplicationLoadBalancer/ApplicationLoadBalancerStack.cs